src/mutation.ts

```diff
@@ -116,6 +116,11 @@ export class Mutation {
 
       if (Number.isSafeInteger(num)) {
         return num;
+      } else {
+        const dblVal = buf.readDoubleBE();
+        if (Number(dblVal) === dblVal && dblVal % 1 !== 0) {
```

Comment:
That is a mistake. The issue here is if I simply do a Number and NaN check, an 8 byte string (eg 'abcdefgh') can pass for a double if the flag is set. 
-        if (Number(dblVal) === dblVal && dblVal % 1 !== 0) {
+        if (!Number.isNaN(Number(dblVal)) && Number(dblVal) === dblVal) {

Test:
       it('should not convert a base64 encoded string if the size is 8 bytes', () => {
         const encoded = Buffer.from('abcdefgh').toString('base64');
         const decoded = Mutation.convertFromBytes(encoded, {isPossibleNumber: true,});
 
         assert.strictEqual('abcdefgh', decoded);
       });

+ 'abcdefgh'
- 1.2926117907728089e+161

I think it would be better if we make the default behaviour to return a Buffer and provide helper functions to return string, float etc.

Reply:
'should not convert a base64 encoded string if the size is 8 bytes', 

Shouldn't we though? We don't know that the value stored was ` 'abcdefgh' `or `1.2926117907728089e+161` and the user indicated the value may be a number. So, unless we failed to convert, it is a number.

Reply:
Makes sense. Updated the PR